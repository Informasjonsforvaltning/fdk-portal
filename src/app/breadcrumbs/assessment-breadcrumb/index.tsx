import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';
import translations from '../../../lib/localization';

import withAssessment, {
  Props as AssessmentProps
} from '../../../components/with-assessment';

interface Props extends AssessmentProps {}

const AssessmentBreadcrumb: FC<Props> = ({ assessment }) => (
  <>
    {translate(assessment?.entity?.title) ?? translations.breadcrumb.notFound}
  </>
);

export default compose<FC>(memo, withAssessment)(AssessmentBreadcrumb);
