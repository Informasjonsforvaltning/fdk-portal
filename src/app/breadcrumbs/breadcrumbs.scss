@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-p-path {
  i, span {
    @include themify($themes) {
      color: themed('color-neutral-dark') !important;
    }
  }
  a {
    @include themify($themes) {
      color: themed('color-text-link') !important;
    }
  }
}
.fdk-fa-path {
  margin: 0 .5em;
  font-size: 1.4rem !important;
}
