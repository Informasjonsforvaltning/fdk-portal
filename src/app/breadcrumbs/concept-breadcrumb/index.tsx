import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';
import translations from '../../../lib/localization';

import withConcept, {
  Props as ConceptProps
} from '../../../components/with-concept';

interface Props extends ConceptProps {}

const ConceptBreadcrumb: FC<Props> = ({ concept }) => (
  <span>
    {translate(concept?.prefLabel ?? translations.breadcrumb.notFound)}
  </span>
);

export default compose<FC>(memo, withConcept)(ConceptBreadcrumb);
