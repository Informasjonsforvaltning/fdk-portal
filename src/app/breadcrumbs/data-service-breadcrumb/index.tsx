import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';
import translations from '../../../lib/localization';

import withDataService, {
  Props as DataServiceProps
} from '../../../components/with-data-service';

interface Props extends DataServiceProps {}

const DataServiceBreadcrumb: FC<Props> = ({ dataService }) => (
  <span>
    {translate(dataService?.title ?? translations.breadcrumb.notFound)}
  </span>
);

export default compose<FC>(memo, withDataService)(DataServiceBreadcrumb);
