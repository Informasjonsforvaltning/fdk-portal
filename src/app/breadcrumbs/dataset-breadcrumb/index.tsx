import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';
import translations from '../../../lib/localization';

import withDataset, {
  Props as DatasetProps
} from '../../../components/with-dataset';

interface Props extends DatasetProps {}

const DatasetBreadcrumb: FC<Props> = ({ dataset }) => (
  <span>{translate(dataset?.title ?? translations.breadcrumb.notFound)}</span>
);

export default compose<FC>(memo, withDataset)(DatasetBreadcrumb);
