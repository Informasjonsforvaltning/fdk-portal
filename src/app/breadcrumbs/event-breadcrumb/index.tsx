import React, { memo, FC } from 'react';
import { compose } from 'redux';

import withEvent, { Props as EventProps } from '../../../components/with-event';

import { getTranslateText as translate } from '../../../lib/translateText';

interface Props extends EventProps {}

const EventBreadcrumb: FC<Props> = ({ event }) => (
  <span>{translate(event?.title)}</span>
);

export default compose<FC>(memo, withEvent)(EventBreadcrumb);
