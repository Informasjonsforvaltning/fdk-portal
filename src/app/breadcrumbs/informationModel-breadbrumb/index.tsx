import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';

import withInformationModel, {
  Props as InformationModelProps
} from '../../../components/with-information-model';

interface Props extends InformationModelProps {}

const InformationModelBreadcrumb: FC<Props> = ({ informationModel }) => (
  <span>{translate(informationModel?.title)}</span>
);

export default compose<FC>(
  memo,
  withInformationModel
)(InformationModelBreadcrumb);
