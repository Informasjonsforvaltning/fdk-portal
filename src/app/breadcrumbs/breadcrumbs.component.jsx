import React from 'react';
import { NavLink } from 'react-router-dom';
import withBreadcrumbs from 'react-router-breadcrumbs-hoc';

import {
  PATHNAME_CONCEPTS,
  PATHNAME_CONCEPTS_COMPARE,
  PATHNAME_INFORMATIONMODELS
} from '../../constants/constants';
import { DatasetBreadcrumb } from './dataset-breadcrumb/dataset-breadcrumb';
import { ApiBreadcrumb } from './api-breadcrumb/api-breadcrumb';
import { ConceptBreadcrumb } from './concept-breadcrumb/concept-breadcrumb';
import { InformationModelBreadcrumb } from './informationModel-breadbrumb/information-breadcrumb';
import { PathNameBreadcrumb } from './pathname-breadcrumb/pathname-breadcrumb';
import './breadcrumbs.scss';
import { getConfig } from '../../config';

// define some custom breadcrumbs for certain routes (optional)
const routes = [
  {
    path: '/',
    breadcrumb: (
      <PathNameBreadcrumb
        pathName={getConfig().themeNap ? 'homeNAP' : 'home'}
      />
    )
  },
  { path: '/datasets/:id', breadcrumb: DatasetBreadcrumb },
  { path: '/apis/:id', breadcrumb: ApiBreadcrumb },
  {
    path: `${PATHNAME_CONCEPTS}${PATHNAME_CONCEPTS_COMPARE}`,
    breadcrumb: <PathNameBreadcrumb pathName="conceptsCompare" />
  },
  { path: '/concepts/:id', breadcrumb: ConceptBreadcrumb },
  {
    path: `${PATHNAME_INFORMATIONMODELS}/:id`,
    breadcrumb: InformationModelBreadcrumb
  },
  { path: '/about', breadcrumb: <PathNameBreadcrumb pathName="about" /> },
  {
    path: '/about-registration',
    breadcrumb: <PathNameBreadcrumb pathName="aboutRegistration" />
  },
  {
    path: '/reports',
    breadcrumb: <PathNameBreadcrumb pathName="reports" />
  }
];

const options = {
  disableDefaults: true,
  excludePaths: []
};

// map & render your breadcrumb components however you want.
// each `breadcrumb` has the props `key`, `location`, and `match` included!
const PureBreadcrumbs = ({ breadcrumbs }) => {
  if (breadcrumbs && breadcrumbs.length > 1) {
    return (
      <div className="col-xs-12">
        <p className="fdk-p-path">
          {breadcrumbs.map((breadcrumb, index) => (
            <span key={breadcrumb.key}>
              {index < breadcrumbs.length - 1 && (
                <>
                  <NavLink to={breadcrumb.props.match.url}>
                    {breadcrumb}
                  </NavLink>
                  <i className="fa fa-angle-right fdk-fa-path" />
                </>
              )}
              {index === breadcrumbs.length - 1 && breadcrumb}
            </span>
          ))}
        </p>
      </div>
    );
  }
  return null;
};

export const Breadcrumbs = withBreadcrumbs(routes, options)(PureBreadcrumbs);
