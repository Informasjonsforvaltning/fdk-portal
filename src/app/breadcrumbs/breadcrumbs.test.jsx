import React from 'react';
import { shallow } from 'enzyme';
import { Breadcrumbs } from './breadcrumbs.component';

test('should render Breadcrumbs correctly', () => {
  const wrapper = shallow(<Breadcrumbs />);
  expect(wrapper).toMatchSnapshot();
});
