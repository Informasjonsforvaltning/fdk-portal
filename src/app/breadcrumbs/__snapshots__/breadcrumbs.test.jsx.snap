// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Breadcrumbs correctly 1`] = `
<Route>
  <Component />
</Route>
`;
